File: 283.move-zeroes.py
#
# @lc app=leetcode id=283 lang=python3
#
# [283] Move Zeroes
#

from typing import *
# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # 操作数是最少吗?
        left = right = 0
        while right < len(nums):
            if nums[left] == 0:
                nums[left] = nums[right]
                right += 1
            else:
                left += 1
        n
                            
# @lc code=end

File: 283.move-zeroes.go
/*
 * @lc app=leetcode id=283 lang=golang
 *
 * [283] Move Zeroes
 */
package main

//
// @lc code=start
func moveZeroes(nums []int) {
	if len(nums) <= 1 {
		return
	}
	i := 0
	for ; i < len(nums); i++ {
		if nums[i] == 0 {
			break
		}
	}
	if i == len(nums)-1 {
		return
	}
	j := i + 1
	for ; j < len(nums); j++ {
		if nums[j] != 0 {
			nums[i] = nums[j]
			i++
		}
	}
	for ; i < len(nums); i++ {
		nums[i] = 0
	}
}

// @lc code=end
